politeiawww/politeiawww.go

```diff
@@ -72,7 +71,7 @@ type politeiawww struct {
 	auth      *mux.Router // CSRF protected subrouter
 	politeiad *pdclient.Client
 	http      *http.Client // Deprecated; use politeiad client
-	mail      *mail.Client
+	mail      mailer
```

Comment:
> it now requires a user to exist in the db to send an email

the idea was:
- `cockroachdb.FetchHistories24h` will return `empty`/`nil` slice of histories for non-existing user
- `Limiter.SendTo` will just mark such recipients as `good` and will send them an email

but now, since you've mentioned it, I've noticed that `cockroachdb.RefreshHistories24h` will also be called for such users and that's where we start tracking email history for these users as well (even if it is `sysadmin`) ...

Well, I don't recall this to be a explicit requirement (that we need to be able to rate limit for some users, but not the others),
can you clarify which of "send-email" calls must be rate limited, and which mustn't ?

Reply:
The reason this was not discussed previously is because it didn't occur to me until I saw the full implementation.

Bascially, we want a method that includes rate limiting and one that does not.  If being rate limited, a user must exist in the database.  Something along the lines of:

```
// Mailer provides methods for sending emails.
type Mailer interface {
	// IsEnabled returns whether the email server is enabled.
	IsEnabled() bool

	// SendTo sends an email to a list of email addresses. These emails
	// do not have to correspond to politeiawww user emails. This
	// method can be used for alternative use cases like sending
	// notifications to sysadmins.
	SendTo(subject, body string, recipients []string) error

	// SendToUser sends an email to a list of politeiawww users. This
	// method rate limits the number of emails that can be sent to each
	// user.
	//
	// The provided recipients map is a map[uuid]email.
	SendToUser(subject, body string, recipients map[string]string) error
}
```